src/resolver/descriptor.cc

```diff
@@ -457,6 +457,69 @@ to_json( nlohmann::json & jto, const ManifestDescriptorRaw & descriptor )
     }
 }
 
+/* -------------------------------------------------------------------------- */
+
+ManifestDescriptorRaw::ManifestDescriptorRaw(
+  const std::string_view & descriptor )
+{
+  size_t cursor = 0;
+  // Grab the input if it exists
+  if ( auto inputSepIdx
+       = descriptor.find_first_of( ManifestDescriptorRaw::inputSigil );
+       inputSepIdx != std::string_view::npos )
+    {
+      this->packageRepository
+        = std::string( descriptor.substr( cursor, inputSepIdx - cursor ) );
+      cursor = inputSepIdx + 1;
+    }
+  // Grab the attribute path or package name
+  size_t attrsEndIdx;
+  bool   hasVersion = false;
+  if ( auto versionSepIdx
+       = descriptor.find_first_of( ManifestDescriptorRaw::versionSigil );
+       versionSepIdx != std::string_view::npos )
+    {
+      attrsEndIdx = versionSepIdx;
+      hasVersion  = true;
+    }
+  else { attrsEndIdx = descriptor.size(); }
+  auto maybeAttrs
+    = std::string( descriptor.substr( cursor, attrsEndIdx - cursor ) );
+  auto numAttrSeparators
+    = std::count_if( maybeAttrs.begin(),
+                     maybeAttrs.end(),
+                     []( auto attrChar ) { return attrChar == '.'; } );
+  switch ( numAttrSeparators )
+    {
+      case 0:
+        // We were given a `pname`
```

Comment:
Blocking, see https://github.com/flox/pkgdb/blob/ae947b65b46d2d4c6d08ff99b16e79e8a9fd0b97/include/flox/resolver/descriptor.hh#L48
```suggestion
        // Match against `name`, `pname`, or `attrName`
```